src/main/java/com/example/wdmsystem/reservation/system/ReservationService.java

```diff
@@ -35,8 +35,14 @@ public void updateReservation(int reservationId, ReservationDTO request) {
         // TODO: Fetch the reservation, update fields, and save back to repository
     }
 
-    public void deleteReservation(int reservationId) {
-        // TODO: Delete the reservation from the repository
+    public void cancelReservation(int reservationId) {
+        Reservation reservation = reservationRepository.findById(reservationId)
+                .orElseThrow(() -> new RuntimeException("Reservation not found"));
```

Comment:
Use NotFoundException from our custom exceptions instead of RuntimeException, cause this wont get caught by the global exception handler.